bokehjs/src/lib/models/glyphs/webgl/markers.ts

```diff
@@ -87,15 +85,34 @@ export abstract class MarkerGL extends BaseGLGlyph {
   protected vbo_bg_color: VertexBuffer & {used?: boolean}
   protected index_buffer: IndexBuffer
 
-  protected last_trans: Transform
-
-  protected _baked_offset: [number, number]
+  static is_supported(marker_type: MarkerType): boolean {
+    switch (marker_type) {
+      case "asterisk":
+      case "circle":
+      case "circle_cross":
+      case "circle_x":
+      case "cross":
+      case "diamond":
+      case "diamond_cross":
+      case "hex":
+      case "inverted_triangle":
+      case "square":
+      case "square_cross":
+      case "square_x":
+      case "triangle":
+      case "x":
+        return true
+      default:
+        return false
+    }
+  }
 
-  protected init(): void {
-    const {gl} = this
+  constructor(gl: WebGLRenderingContext, readonly glyph: MarkerLikeView, readonly marker_type: MarkerType) {
+    super(gl, glyph)
 
+    const defs = [`#define USE_${marker_type.toUpperCase()}`]
     const vert = vertex_shader
-    const frag = fragment_shader(this._marker_code)
+    const frag = `${defs.join("\n")}\n\n${fragment_shader}`
```

Comment:
I don't love this with the `#define` Why not store the fragment shaders as an attribute on the marker class? Then the concerns are grouped together, the giant every-fragment file can be avoided, and the presence or absence of a fragment shader on a class can directly be used to determine gl-support, rather than having to maintain a separate list in sync, as above?

Reply:
My goal with this was to make it easier to work with shaders, which means allow to use glsl-capable tooling, validation, syntax highlighting, etc. This is currently the best option, not ideal though. We could use `#import` directive, but it's not widely supported, so we would need some regex search and replace to allow that. I considered this, but decided that at this point it was a too expensive solution. Then there are no marker models anymore, so there is no intrinsic place to put those implementations.